content/browser/ml/webnn/dml/graph_dml_impl.cc

```diff
@@ -1840,44 +1829,47 @@ bool GraphDMLImpl::Build(ModelInfoPtr model_info, BuildResult* out_result) {
 
   execution_context_->InitializeGraph(this, mCompiledOperator.Get(),
                                       input_binding_desc);
-
   execution_context_->Flush();
-  execution_context_->WaitForSignal();
-  execution_context_->ReleaseCompletedResources();
 
+  execution_context_->WaitForSignal(
+      base::BindOnce(&GraphDMLImpl::OnWaitForBuildSignal, base::Unretained(this),
+                     std::move(callback)));
+}
+
+void GraphDMLImpl::OnWaitForBuildSignal(BuildCallback callback) {
+  execution_context_->ReleaseCompletedResources();
   auto& named_outputs = graph_desc_builder_->GetNamedOutputs();
   HRESULT hr = output_resource_readback_->InitializeResource(named_outputs);
   if (FAILED(hr)) {
-    *out_result = BuildResult::kUnknownError;
-    return false;
+    std::move(callback).Run(BuildResult::kUnknownError);
+    return;
   }
-  *out_result = BuildResult::kOk;
-  return true;
+  std::move(callback).Run(BuildResult::kOk);
 }
 
 void GraphDMLImpl::Compute(NamedResourcesPtr named_inputs,
                            ComputeCallback callback) {
   TRACE_EVENT0("gpu", "GraphDMLImpl::Compute");
   ExecutionResources* execution_resources =
       execution_context_->GetExecutionResources();
-  ID3D12Resource* inputs_resource =
+   ID3D12Resource* input_resource =
```

Comment:
```suggestion
  ID3D12Resource* input_resource =
```

[nit] Indent is misaligned now with other lines.